testsuite/classlibrary/TestReadableNodeIDAllocator.sc

```diff
@@ -0,0 +1,85 @@
+TestReadableNodeIDAllocator : UnitTest {
+
+	// test proper creation from Server
+	test_readableNodeRange {
+		var s = Server(\testRNIA);
+		var prevClass = Server.nodeAllocClass;
+		Server.nodeAllocClass = ReadableNodeIDAllocator;
+		s.options.maxLogins = 1; s.newAllocators;
+		// this.assert(
+		// 	(s.nodeAllocator.numIDs == (10 ** 8)),
+		// 	"for a single client, (readable) nodeAllocator has maximum range."
+		// );
+		// this.assert(
+		// 	(s.nodeAllocator.numIDs == (10 ** 7)),
+		// 	"for 16 clients, (readable) nodeAllocator has divided range."
+		// );
+
+		// same for ReadableNodeIDAllocator
+		// Server.nodeAllocClass = ReadableNodeIDAllocator;
+		Server.named.removeAt(s.name); Server.all.remove(s);
+		Server.nodeAllocClass = prevClass;
+	}
+
+	test_permIDs {
+		var alloc = ReadableNodeIDAllocator(8191, 1000, 8191);
+		var nextPermID, permIDs = List[], permIDToFree;
+
+		this.assert(
+			alloc.isPerm(alloc.idOffset - 1).not
+			and: alloc.isPerm(alloc.idOffset)
+			and: alloc.isPerm(alloc.idOffset + alloc.lowestTempID - 1)
+			and: alloc.isPerm(alloc.idOffset + alloc.lowestTempID).not,
+			"ReadableNodeIDAllocator should identify its permanent IDs."
+		);
+		while {
+			if (0.5.coin and: { permIDs.size > 0 }) {
```

Comment:
please avoid randomness in tests (it makes it hard/impossible to reproduce a failing case)

Reply:
this is now fixed with the random seed.